moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move

```diff
@@ -166,20 +166,26 @@ module moveos_std::account_storage {
         let account_storage = borrow_account_storage(storage_context::object_storage(ctx), account);
         exists_module_at_account_storage(account_storage, name) 
     }
-
     /// Publish modules to the account's storage
-    public fun publish_modules(ctx: &mut StorageContext, account: &signer, modules: vector<MoveModule>) {
+    public fun publish_modules(_ctx: &mut StorageContext, account: &signer, modules: vector<MoveModule>) {
         let account_address = signer::address_of(account);
-        let account_storage = borrow_account_storage_mut(storage_context::object_storage_mut(ctx), account_address);
         let i = 0;
         let len = vector::length(&modules);
-        let module_names = move_module::verify_modules(&modules, account_address);
+        let module_bytes = vector::empty<vector<u8>>();
+        
         while (i < len) {
-            let name = vector::pop_back(&mut module_names);
             let m = vector::pop_back(&mut modules);
-            table::add(&mut account_storage.modules, name, m);
-        }
+            vector::push_back(&mut module_bytes, move_module::module_bytes(m));
+            i = i + 1;
+        };
+        request_publish(account_address, module_bytes);
```

Comment:
Why not directly write the module bytes to the module table?

Reply:
Fow now, verifying a compiled module requires a `ModuleResolver`. I havn't find a way to verify module in a native context.

Reply:
We can refactor the NativeContext. The `DataStore` in NativeContext can be a `ModuleResolver`.

Reply:
OK， I will try this way.

Reply:
> We can refactor the NativeContext. The `DataStore` in NativeContext can be a `ModuleResolver`.

The `ModuleResolver` need specific `Error` type. Refactoring `DataStore`  could potentially have a significant impact. This may not be a suitable way.